ebcl/tools/hypervisor/vbus_gen.py

```diff
@@ -0,0 +1,200 @@
+
+            base_address = 0
+            size = 0
+
+            # Extract address (1 or 2 cells)
+            if address_cell == 1:
+                base_address = int(reg_values[i], 16)
+            elif address_cell == 2:
+                base_address = (int(reg_values[i], 16) << 32) + int(reg_values[i + 1], 16)
+
+            # Extract size (1 or 2 cells)
+            if size_cell == 1:
+                size = int(reg_values[i + address_cell], 16)
+            elif size_cell == 2:
+                size = (int(reg_values[i + address_cell], 16) << 32) + int(reg_values[i + address_cell + 1], 16)
+
+            mmios.append({"address": f"0x{base_address:x}", "size": f"0x{size:x}"})
+
+        return mmios
+
+    def parse_interrupts_property(self, interrupts_property) -> list:
+        """Parse the interrupts property and return a list of dictionaries with IRQ numbers."""
+        interrupts_cleaned = interrupts_property.replace('<', '').replace('>', '').replace(';', '').split()
+        irqs: list = []
+        
+        if len(interrupts_cleaned) == 1:
+            logging.warning(f'TODO: Only one interrupt entry: {interrupts_cleaned[0]}')
+            return irqs
+
+        for i in range(0, len(interrupts_cleaned), 3):
+            interrupt_id = int(interrupts_cleaned[i + 1], 16)  # Second value in the group is the interrupt ID
+            irq_number = 32 + interrupt_id  # Convert interrupt ID to IRQ number
+            irqs.append({"irq": irq_number, "trigger": "level_high"})
+
+        return irqs
+    
+    def parse_cells_info(self, cell_property) -> int:
+        cell_cleaned = cell_property.replace('<', '').replace('>','').replace(';','')
+        cell = int(cell_cleaned, 16)
+        return cell
+    
+    def parse_node_full_name(self, node_match: re.Match[str]) -> str:
+        node_name, unit_address = node_match.groups()
+        full_name = node_name if not unit_address else f"{node_name}@{unit_address}"
+        return full_name
+    
+
+    def process_dts_file(self, dts_file_path):
+        """Parse a DTS file and convert it into the YAML structure."""
+        try:
+            with open(dts_file_path, 'r') as dts_file:
+                dts_lines = dts_file.readlines()
+            
+            # Regex patterns to extract nodes and properties
+            node_pattern = re.compile(r'([a-zA-Z0-9\-_]+)\s*@?([\da-fA-Fx]*)\s*{')
+            compatible_pattern = re.compile(r'compatible\s*=\s*(\"[^\"]+\")\s*;')
+            reg_pattern = re.compile(r'reg\s*=\s*(<[^>]+>)')
+            interrupts_pattern = re.compile(r'interrupts\s*=\s*(<[^>]+>)')
+            status_pattern = re.compile(r'status\s*=\s*(\"?[a-zA-Z0-9_]+\"?)\s*;')
+            size_cell_pattern = re.compile(r'#size-cells\s*=\s*<\s*(0x[0-9a-fA-F]+)\s*>')
+            address_cell_pattern = re.compile(r'#address-cells\s*=\s*<\s*(0x[0-9a-fA-F]+)\s*>')
+            end_node_pattern = re.compile(r'}')  # Matches closing brace of a node
```

Comment:
This will do more harm than good in the end.

To give you an example where this will fail horrible:
Take the original device tree (not the decompiled version) and preprocess it, this will yield interrupt entries like
```
interrupts = <0x00 0x80 0x04>,
                    <0x00 0x80 0x04>, 
                    <0x00 0x81 0x04>, 
                    ...;
```
This is the canonical form used in device trees to define interrupts. The decompiler doesn't interpret the #cell-count and so decompiles them as one big list of values.

The regexp used here will only pickup the first irq. Same thing for `regs` by the way